pkgs/development/python-modules/flax/default.nix

```
@@ -0,0 +1,60 @@
+{ buildPythonPackage
+, fetchFromGitHub
+, keras
+, lib
+, matplotlib
+, msgpack
+, numpy
+, optax
+, pytestCheckHook
+, tensorflow
+}:
+
+buildPythonPackage rec {
+  pname = "flax";
+  version = "0.3.6";
+
+  src = fetchFromGitHub {
+    owner = "google";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0zvq0vl88hiwmss49bnm7gdmndr1dfza2bcs1fj88a9r7w9dmlsr";
+  };
+
+  propagatedBuildInputs = [
+    matplotlib
+    msgpack
+    numpy
+    optax
+  ];
+
+  pythonImportsCheck = [
+    "flax"
+  ];
+
+  checkInputs = [
+    keras
+    pytestCheckHook
+    tensorflow
+  ];
+
+  disabledTestPaths = [
+    # Docs test, needs extra deps + we're not interested in it.
+    "docs/_ext/codediff_test.py"
+
+    # The tests in `examples` are not designed to be executed from a single test
+    # session and thus either have the modules that conflict with each other or
+    # wrong import paths, depending on how they're invoked. Many tests also have
+    # dependencies that are not packaged in `nixpkgs` (`clu`, `jgraph`,
+    # `tensorflow_datasets`, `vocabulary`) so the benefits of trying to run them
+    # would be limited anyway.
+    "examples/*"
+  ];
+
+  meta = with lib; {
+    description = "Flax is a neural network library for JAX designed for flexibility.";
```

See https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes.

```suggestion
    description = "Neural network library for JAX";
```

I suggest to skip the "marketing" parts. For upstream it's "flexibility" for the average user perhaps "complexity".

---

I typically just use the first sentence from the "Description / Overview" section in README to let the authors describe their package in the words they prefer. But no strong preferences from my side, changed to your suggestion.